kinto_http/session.py

```diff
@@ -100,22 +100,31 @@ def request(self, method, endpoint, data=None, permissions=None, payload=None, *
         kwargs["headers"] = {"User-Agent": USER_AGENT, **self.headers, **overridden_headers}
 
         payload = payload or {}
+
+        if method.lower() == "get" and data is not None:
+            raise KintoException("GET requests are not allowed to have a body!")
+
         if data is not None:
             payload["data"] = data
+
         if permissions is not None:
             if hasattr(permissions, "as_dict"):
                 permissions = permissions.as_dict()
+
             payload["permissions"] = permissions
-        if method not in ("get", "head"):
+
+        if method.lower() not in ("get", "head"):
             if "files" in kwargs:
                 kwargs.setdefault("data", payload)
+
             else:
                 kwargs.setdefault("data", utils.json_dumps(payload))
                 kwargs["headers"].setdefault("Content-Type", "application/json")
```

Comment:
good point, moving this conditional down makes it obsolete, though, because the fixed `lower()` in line 113 prevents the setdefault from being executed.